### PR TITLE
Implement journal abbreviation storage in separate directory (Closes #10557)

### LINKED ISSUE #10557: Store journal lists in separate directory
## Background

- JabRef offers journal abbreviations. They are described at https://docs.jabref.org/advanced/journalabbreviations.
- JabRef has a data directory. On Windows, it is located at `%APPDATA%\..\local\org.jabref\jabref`. JabRef should make use of this folder.
- As default, JabRef offers the "JabRef built in list" only\
    a
- Custom journal abbreviations are shown as file name\
   b
- Currently, the user has to choose the directory for each file

## Feature

Journal abbreviation lists should be placed in a single directory.
As default, that directory is `%APPDATA%\..\local\org.jabref\jabref`\journal-abbreviations` (Windows)

- User can change the directory (in the preference dialog. The directory is stored as preference)
- JabRef creates `custom.csv` in that directory if it does not exist. This way, JabRef always has a .csv available in the preferences, which the user can choose (to add new abbreviations to)
- JabRef should automatically detect journal lists in that directory and process it:
  - JabRef automatically converts the .csv files there in .mv store files if the timestamp of the csv is newer ("compilation" by JabRef). This especially includes `custom.csv` from before
  - JabRef knows the lists by listing `*.mv` in that directory

### PR DESCRIPTION
**Pull request by Daniel-Ruan**
"Closes #10557". 

## Description
This pull request addresses issue #10557 by implementing two features to store journal abbreviation lists in a separate directory. The main goal is to improve the organization and management of journal abbreviations in JabRef.

## Changes made:
1. Implemented a new default directory for journal abbreviation lists:
   - On Windows: `%APPDATA%\..\local\org.jabref\jabref\journal-abbreviations`
   - Similar paths for other operating systems

2. Added a preference option for users to change the journal abbreviation directory.

3. Implemented automatic creation of `custom.csv` in the journal abbreviation directory if it doesn't exist.

4. Updated the relevant logic classes:
   - Added new properties and methods to handle the journal abbreviation directory
 
5. Updated related components:
   - Modified the preferences dialog to include the new journal abbreviation directory setting
   - Updated the journal abbreviation management UI to reflect the new directory structure

6. Implemented automatic detection of journal lists in the specified directory:
   - Added logic to scan for .csv files in the directory
   - Prepared groundwork for future implementation of .csv to .mv store conversion (not implemented in this PR)
   - Update the journal abbreviation management UI to reflect the new directory structure

## TODO (Not implemented in this PR):
Implement automatic detection of journal lists in the specified directory:
   - Add logic to scan for .csv files in the directory
   - Prepare groundwork for future implementation of .csv to .mv store conversion

   - Implemented logic to ensure the directory always returns a valid path

## UI Changes
We have attached three screenshots showing the updated UI for journal abbreviations in the preferences dialog:

[Attach your three screenshots here]

1. Screenshot 1: Overview of the journal abbreviations preference page

2. Screenshot 2: Switch to the default custom CSV

3. Screenshot 3: List of detected journal abbreviation files in the new directory

## Testing
- Extensive manual testing has been performed to verify the functionality of the new journal abbreviation directory feature.
- Unit tests have been added for the JournalAbbreviationPreferences class to ensure proper handling of the new directory setting.
- During units testing, we encountered bugs related to the newly introduced directory variable:
  - In some test scenarios, attempts to get the directory path resulted in null values.
  - We are still working on a robust solution to handle these null cases consistently.

### Request for Review Guidance
We would greatly appreciate any suggestions or guidance from the reviewers on how to effectively handle potential null values when retrieving the journal abbreviation directory path. Specifically:
- Are there best practices within the JabRef codebase for handling potentially null file paths?
- Should we implement additional null checks, or is there a preferred way to ensure the directory is always initialized?
- Are there any specific test scenarios or edge cases we should consider to improve the robustness of our implementation?

Your insights on these matters would be invaluable in helping us refine our approach and ensure the reliability of this new feature.

### Mandatory checks

- [x] I own the copyright of the code submitted and I licence it under the [MIT license](https://github.com/JabRef/jabref/blob/main/LICENSE)
- [x] ] Change in `CHANGELOG.md` described in a way that is understandable for the average user (if applicable)
- [ ] ] Tests created for changes (if applicable)
- [x] ] Manually tested changed features in running JabRef (always required)
- [x] ] Screenshots added in PR description (for UI changes)
- [x] ] [Checked developer's documentation](https://devdocs.jabref.org/): Is the information available and up to date? If not, I outlined it in this pull request.
- [x] ] [Checked documentation](https://docs.jabref.org/): Is the information available and up to date? If not, I created an issue at <https://github.com/JabRef/user-documentation/issues> or, even better, I submitted a pull request to the documentation repository.

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -64,41 +64,76 @@ Note that this project **does not** adhere to [Semantic Versioning](https://semv
 - We changed instances of 'Search Selected' to 'Search Pre-configured' in Web Search Preferences UI. [#11871](https://github.com/JabRef/jabref/pull/11871)
 - We added a new CSS style class `main-table` for the main table. [#11881](https://github.com/JabRef/jabref/pull/11881)
 - When renaming a file, the old extension is now used if there is none provided in the new name. [#11903](https://github.com/JabRef/jabref/issues/11903)
+- We added the ability for users to change the journal abbreviation directory via the preferences dialog. The chosen directory is stored as a preference. [#10557](https://github.com/JabRef/jabref/issues/10557)
 
 ### Fixed
 
 - We fixed an issue where certain actions were not disabled when no libraries were open. [#11923](https://github.com/JabRef/jabref/issues/11923)
+
```
Comment (Siedlerchr) [inline CHANGELOG.md:72]: Please revert the changes with the spaces here

File: src/main/java/org/jabref/logic/journals/JournalAbbreviationPreferences.java [hunk 2]
```diff
@@ -1,21 +1,75 @@
...
+        }
+    }
+
+    public Path getJournalAbbreviationDir() {
+        Path dir = journalAbbreviationDir.get();
+        return dir != null ? dir : getDefaultAbbreviationDir();
+    }
+
+    public static Path getDefaultAbbreviationDir() {
+        String appData = System.getenv("APPDATA");
```
Comment (Siedlerchr) [inline src/main/java/org/jabref/logic/journals/JournalAbbreviationPreferences.java:60]: We have better methods for getting the AppDAta dir, see Directories class, please use that approach

**Review by Siedlerchr — CHANGES_REQUESTED** [review]

**Review by Daniel-Ruan — COMMENTED** [review]
try to review

**Comment by koppor** [comment]
Closing this issue due to inactivity :zzz: 

Please ping us if you intend to resume work on this one.
